### ISSUE TITLE
required: true does not work properly with date fields.

### ISSUE THREAD

**Issue by peterboorsma** [bolt/core#1743]
When adding the required: true option to a date field causes Bolt to behave unexpectedly. The date editor keeps popping up when trying to save the record.
See [this screencast](https://www.dropbox.com/s/wrgr1r3ln2e6j3b/Screen%20Recording%202020-08-20%20at%2017.49.41.mov?dl=0) for an example: 

My contenttype:

```
        date_to:
            type: date
            default: "first day of this month"
            required: true
```

---
Bolt 4.0.0. RC35

**Comment by eduardomart** [comment]
Is this meant to be fixed in RC41? I am still having problems with it
<img width="739" alt="Screen Shot 2020-09-02 at 4 54 14 PM" src="https://user-images.githubusercontent.com/1833020/92048745-5e5a8500-ed3d-11ea-93ca-7a315323e2ae.png">

Issue occurs in Firefox and Safari

**More info:**
My date field has required: true but I can't save, no matter what date I pick

**Possible temp workaround:**
Once you select a date from the picker, the input lets you type a date manually, if you type something like: mm/dd/yyyy the record can be saved

**Comment by eduardomart** [comment]
@peterboorsma is this working correctly for you? it seems it's still an issue on 4.1.4 for me, I can pick a date when I first create a record, but if later I come back and clear it and pick a new date, I can't never save the record again, unless I do the workaround I mentioned above.

**Comment by peterboorsma** [comment]
The problem does not seem to be solved in Bolt 4.1.4. It shows exactly the same behaviour as described initially in this issue.

**Comment by peterboorsma** [comment]
Reopened because the issue isn't solved...
